internal/services/v0/sharestore.go

```diff
@@ -158,16 +151,16 @@ func (s3s *s3ShareStore) LookupSharedByReference(reference string) (SharedDataV2
 
 	ctx := context.Background()
 
-	result, err := s3s.s3Client.GetObjectWithContext(ctx, &s3.GetObjectInput{
+	result, err := s3s.s3Client.GetObject(ctx, &s3.GetObjectInput{
 		Bucket: aws.String(s3s.bucket),
 		Key:    aws.String(key),
 	})
 	if err != nil {
-		var aerr awserr.Error
-		if errors.As(err, &aerr) && aerr.Code() == s3.ErrCodeNoSuchKey {
+		var nsk *types.NoSuchKey
+		if errors.As(err, &nsk) {
```

Comment:
if the type is a pointer, should this be a direct reference (sans &)?

Reply:
No, based on the example here: https://pkg.go.dev/errors#example-As

it shouldn't